cpp/src/parquet/column_reader.cc

```diff
@@ -1151,6 +1159,14 @@ int64_t TypedColumnReaderImpl<DType>::ReadBatchSpaced(
   return total_values;
 }
 
+template <typename DType>
+void TypedColumnReaderImpl<DType>::InitScratchForSkip(int64_t batch_size) {
```

Comment:
What if the batch size is not the same between calls?

Reply:
I removed this argument and clarified this in the code. The batch size is constant and will not change.